include/hobbes/lang/preds/subtype/obj.H

```diff
@@ -20,7 +21,7 @@ namespace hobbes {
 //   * the constant adjustment    (for A <: B, A : ..., B;    the memory for B is at an offset from A)
 //   * the vtbl lookup adjustment (for A <: B, A : virtual B; A holds a pointer to the base address for B)
 struct PtrAdjustment {
-  PtrAdjustment(bool vtblLookup, int offset, const std::string& targetTy) : vtblLookup(vtblLookup), offset(offset), targetTy(targetTy) { }
+  PtrAdjustment(bool vtblLookup, int offset, std::string targetTy) : vtblLookup(vtblLookup), offset(offset), targetTy(std::move(targetTy)) { }
```

Comment:
what's the justification for this change?

Reply:
For the argument `targetTy`, the number is the number of constructor calls

| param type \ arg type| const char* | std::string (lvalue) | std::string (rvalue) |
|-|-|-|-|
|const std::string&| 2  | 1 | 1  |
|std::string + std::move | 1  | 1 | 0 |

Overall, this idiom gives reasonable better performance without adding extra constructor overloads or resort to templates